ENIGMAsystem/SHELL/Universal_System/Resources/polygon_internal.h

```diff
@@ -0,0 +1,142 @@
+/** Copyright (C) 2021 Nabeel Danish
+*** Nabeel Danish
+**/
+
+#ifndef ENIGMA_POLYGON_INTERNAL_H
+#define ENIGMA_POLYGON_INTERNAL_H
+
+#include <vector>
+#include "AssetArray.h"
+
+namespace enigma 
+{
+	#define NULL_POINT -1;
+	#define precision pow(4, -10)
+
+	// The Point class represents a point in a polygon
+	// Resource
+	class Vector2D 
+	{
+		// Attributes
+		private: 
+			double x;
+        	double y;
+
+		// Methods
+		public:
+			Vector2D();
+			Vector2D(double x, double y);
+			Vector2D(const Vector2D& obj);
+			void copy(const Vector2D& obj);
+
+			double getX();
+			double getY();
+			double getMagnitude();
+			double getAngle();
+			Vector2D getNormR();
+			Vector2D getNormL();
+			Vector2D getUnitVector();
+
+			void setX(double x);
+			void setY(double y);
+			void setMagnitude(double number);
+			void setAngle(double angle);
+			void scale(double number);
+			void invert(bool x, bool y);
+
+			Vector2D add(const Vector2D& B);
+			Vector2D minus(const Vector2D& B);
+			Vector2D rotate(double angle);
+			Vector2D interpolate(double number);
+
+			double angleBetween(Vector2D& B);
+			double dotProduct(const Vector2D& B);
+			double perpProduct(const Vector2D& B);
+			double crossProduct(const Vector2D& B);
+
+			bool equal(const Vector2D& B);
+
+			void print();
+	};
+
+	// The Polygon class represents the polygon that will be used
+	// for detecting collisions in 2D space by the collision detection 
+	// system
+	class Polygon {
+		// Attributes
+		private:
+			std::vector<Vector2D> points;
+			std::vector<std::vector<Vector2D>> subpolygons;
+			int height;
+			int width;
+			int xoffset;
+			int yoffset;
+			int numSubPolygons;
+			double scale;
+			double angle;
```

Comment:
This is something that I guess @fundies mentioned about having polygon transformations per instance instead of per polygon. If I am getting this correct, than this is something that I am working on now